test/integration/resource.ts

```diff
@@ -0,0 +1,41 @@
+import {expect} from 'chai';
+import {Ketting, NeverCache} from '../../src';
+
+describe('Using resources', async () => {
+  const ketting = new Ketting('http://localhost:3000/hal1.json');
+
+  it('concurrent get on the same resource should do 2 distinct HTTP calls if request headers are different', async () => {
+    ketting.cache = new NeverCache();
+
+    const resource = await ketting.follow('collection');
+
+    const [firstState, secondState] =
+      await Promise.all([
+        resource.get({headers: {'delay-in-ms': '2', 'foo': 'bar'}}),
+        resource.get({headers: {'foo': 'baz'}})
+      ]);
+
+    expect(firstState.timestamp).to.not.eql(secondState.timestamp);
+  });
+
+  it('concurrent get on the same resource should produce only one HTTP call if request headers are the same', async () => {
```

Comment:
I don't think this test fails without your fix

Reply:
Yes this is the behavior that was here before the fix.
Removing the test

Reply:
Should this test be removed ?

Reply:
To be more accurate about my remark, the test should fail if your request equality implementation return falsd instead of true while all headers are the same.
I think this test should be kept using unique id per response instead of timestamp.